### PR TITLE
Use iam to get tls config

### PR DESCRIPTION
**Pull request by Eabramov84**

### CODE REVIEW

**Review by MykolaSolyanko — APPROVED** [review]

File: communicationmanager.go [hunk 1]
```diff
@@ -137,13 +140,17 @@ func newCommunicationManager(cfg *config.Config) (cm *communicationManager, err
 		return cm, aoserrors.Wrap(err)
 	}
 
+	if cm.cryptoContext, err = cryptutils.NewCryptoContext(cfg.Crypt.CACert); err != nil {
+		return nil, aoserrors.Wrap(err)
+	}
+
```
Comment (al1img) [inline communicationmanager.go:146]: Also, the initialization of TPM device and PKCS11 library should be moved here from fcrypt.

File: smcontroller/smcontroller.go [hunk 2]
```diff
@@ -90,7 +90,6 @@ type CertificateProvider interface {
 /***********************************************************************************************************************
  * Public
  **********************************************************************************************************************/
-
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply smcontroller/smcontroller.go:88]: Done

File: umcontroller/umctrlserver.go [hunk 3]
```diff
@@ -53,6 +53,7 @@ type umCtrlServer struct {
  **********************************************************************************************************************/
 
 // NewServer create update controller server.
+
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply umcontroller/umctrlserver.go:56]: Done

File: umcontroller/umhandler.go [hunk 4]
```diff
@@ -260,10 +260,12 @@ func (handler *umHandler) sendPrepareUpdateRequest(e *fsm.Event) {
 }
 
 func (handler *umHandler) updateStatusNtf(e *fsm.Event) {
+	status, _ := e.Args[0].(umStatus)
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply umcontroller/umhandler.go:263]: Done

File: communicationmanager.go (hunk 1 above)
Comment (MykolaSolyanko) [reply communicationmanager.go:146]: Done

File: config/config_test.go [hunk 5]
```diff
@@ -171,11 +172,17 @@ func TestGetBoardConfigFile(t *testing.T) {
...
 func TestGetIAMServerURL(t *testing.T) {
-	if testCfg.IAMServerURL != "localhost:8090" {
+	if testCfg.IAMServerURL != "localhost:8089" {
 		t.Errorf("Wrong IAM server value: %s", testCfg.IAMServerURL)
 	}
 }
 
+func TestGetIAMPublicServerURL(t *testing.T) {
+	if testCfg.IAMPublicServerURL != "localhost:8090" {
+		t.Errorf("Wrong IAM server value: %s", testCfg.IAMPublicServerURL)
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply config/config_test.go:182]: according to the documentation 
https://kb.epam.com/display/EPMDAEPRA/Core

**Review by al1img — APPROVED** [review]

**Pull request merged**